mmv1/products/dialogflowcx/terraform.yaml

```
@@ -0,0 +1,41 @@
+# Copyright 2020 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Provider::Terraform::Config
+overrides: !ruby/object:Overrides::ResourceOverrides
+  Agent: !ruby/object:Overrides::Terraform::ResourceOverride
+    timeouts: !ruby/object:Api::Timeouts
+      insert_minutes: 40
+      update_minutes: 40
+    examples:
+      - !ruby/object:Provider::Terraform::Examples
+        name: "dialogflowcx_agent_full"
+        primary_resource_id: "full_agent"
+        vars:
+          agent_name: "dialogflowcx-agent"
+    id_format: "{{name}}"
+    import_format: ["{{name}}"]
+    # Skip sweeper gen due to non-standard urls.
+    skip_sweeper: true
+    properties:
```

```suggestion
    properties:
      name: !ruby/object:Overrides::Terraform::PropertyOverride
        custom_flatten: templates/terraform/custom_flatten/name_from_self_link.erb
```

This function will take a partial URI like `projects/project/locations/global/agents/abcd1234` and only store `abcd1234`. We store the other information in separate fields in Terraform, and the original value from `name` is exported as the implicit `id` field.